javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/HelperInjector.java

```diff
@@ -189,8 +173,60 @@ public static HelperInjector forDynamicTypes(
     return builder;
   }
 
-  private static Map<String, Class<?>> injectBootstrapClassLoader(
-      Map<String, byte[]> classnameToBytes) throws IOException {
+  private ClassLoader injectHelperClasses(
+      TypeDescription typeDescription, ClassLoader classLoader, JavaModule module) {
+    if (classLoader == BOOTSTRAP_CLASSLOADER) {
+      classLoader = BOOTSTRAP_CLASSLOADER_PLACEHOLDER;
+    }
+    if (classLoader == BOOTSTRAP_CLASSLOADER_PLACEHOLDER && instrumentation == null) {
+      log.error(
+          "Cannot inject helpers into bootstrap classloader without an instance of Instrumentation. Programmer error!");
+      return classLoader;
```

Comment:
Is it Ok to throw exceptions here? Or we reasonably expect this should never makes its way to production?

Reply:
I think production usage always sets `instrumentation`? Also no problem to leave as is, this may become clearer in the future if we're able to more clearly separate the dual nature of this class.